### PR TITLE
program setup

### PR DESCRIPTION
**Pull request by rpitingolo**
@maggiesu0725 I started setting up this program. Can you please add the rest of the geographies, test them, and then add a finalize macro inside the %mar_geo macro? Thanks!

### CODE REVIEW

**Comment by rpitingolo** [comment]
@lhendey sorry, tagged you on the wrong pull request before. Do you want to take a final review on this one?

**Comment by lhendey** [comment]
@rpitingolo - yes, but probably have to look tomorrow morning.

**Comment by rpitingolo** [comment]
@lhendey that's fine. Thanks.

**Comment by lhendey** [comment]
@rpitingolo  - not complete - i asked you to review the conversation on the issue which talked about merging on UI_proptype and at least getting residential units vs. not. https://github.com/NeighborhoodInfoDC/MAR/issues/15#issuecomment-403508046 

if status is active doe this match active_res_unit_count? How does active_res_unit_count and active_res_occupancy_count differ?

**Comment by rpitingolo** [comment]
@lhendey the SSLs match between the parcel_base and address point file for all except condo units. If you assume all of the non-matches are condo units then you can split out the counts by type as I've done. I hand checked a bunch and all the addresses corresponded to condos so I think it's a somewhat safe assumption. 

I'm not entirely sure what  active_res_unit_count is. active_res_occupancy_count more closely matches to the ACS count of housing units (307k MAR vs 315k ACS). active_res_unit_count only sums up to about 200k. 

All of my edits are committed for review.

**Comment by lhendey** [comment]
@rpitingolo  - condo unit ssls are in mar.address_ssl_xref

See for example (my building) 
 proc print data=mar.address_ssl_xref;
where ADDRESS_ID=146383;
run;

I think you want to merge the address_ssl_xref with address_points before the merge with parcel base. You could filter out the condo record lots (the building ssls)  using the "lot_type" on the address_ssl_xref. 

Then i'd see how that merge with parcel base works.

File: Prog/MAR_summary_units.sas [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+
+proc sort data = mar.address_points_&address_pt_date. out = addpt_in; by ssl; run;
+proc sort data = realprop.parcel_base out = parcel_in; by ssl; run;
+
+
+/* Set Address Points dataset from most recent update */
+data mar_units;
+	merge addpt_in (in=a)
+		  parcel_in (in=b where=(in_last_ownerpt=1));
```
Comment (lhendey) [inline Prog/MAR_summary_units.sas]: @rpitingolo - i added this restriction - can you chekc - i think it's included when we do the num-units file from ownerpoint.

File: Prog/MAR_summary_units.sas [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+	   a non-match to the address pt file is a condo unit */
+	if a and ui_proptype in (" ") then res_flag = 1;
+
+
+	/* Count number of units by property type*/
+	if res_flag = 1 then do;
+		total_res_units = ACTIVE_RES_OCCUPANCY_COUNT;
+		if ui_proptype in ("10") then total_sf_units = ACTIVE_RES_OCCUPANCY_COUNT;
+		if ui_proptype in ("13") then total_mfapt_units = ACTIVE_RES_OCCUPANCY_COUNT;
+		if ui_proptype in ("11"," ") then total_condo_units = ACTIVE_RES_OCCUPANCY_COUNT;
```
Comment (lhendey) [inline Prog/MAR_summary_units.sas]: @rpitingolo - technically condos could be in single family structures. I took out the multifamily bit. i'd be interested if you can think of how to make the distinction between the two types  - if you look at use code for example condo are distinguished between horizontal and vertical  see file:///L:\Libraries\Realprop\Doc\UseCode%20&%20UI_proptype.xls for the definitions of the usecodes.

Comment (rpitingolo) [reply Prog/MAR_summary_units.sas]: @lhendey couldn't we just use active_res_occupancy_count and if the value is >1 and condo then define it as a MF condo building?

**Comment by rpitingolo** [comment]
@lhendey Thanks. I'll try using mar.address_ssl_xref and see what happens.

File: Prog/MAR_summary_units.sas (hunk 2 above)
Comment (lhendey) [reply Prog/MAR_summary_units.sas]: @rpitingolo no. single-family structures are not necessarily single-family.  Like the rowhouses in capital hill? they aren't getting coded as multifamily. I'm thinking about the rowhouses in columbia heights that are getting renovated an split into 3 unit condominiums.

**Comment by rpitingolo** [comment]
@lhendey using mar.address_ssl_xref for condos seems to work and the total count now matches very closely to ACS for the city (315k units).

File: Prog/MAR_summary_units.sas (hunk 2 above)
Comment (rpitingolo) [reply Prog/MAR_summary_units.sas]: @lhendey I would consider a columbia heights row house with 3 condos to be multi-family though? Maybe we are splitting hairs and should just leave it as you edited.

Comment (lhendey) [reply Prog/MAR_summary_units.sas]: @rpitingolo yes it is a building with multiple families, but the structure type is not multifamily. it's fine not to make that distinction here.

**Comment by lhendey** [comment]
@ptatian  - if we have all the address_points files by date- there isn't a need to save the summary files with the date too, right? We could always generate them for particular dates? Is there anything else you'd want on this summary file before we finalize and merge?

**Comment by lhendey** [comment]
@ptatian = so we should set it up that current files would have _2018 indicators and then when run next year we'd add _2019?

**Comment by ptatian** [comment]
@lhendey Yes.

**Pull request merged**